ivy/functional/ivy/experimental/statistical.py

```diff
@@ -894,3 +896,52 @@ def cummin(
     return ivy.current_backend(x).cummin(
         x, axis=axis, exclusive=exclusive, reverse=reverse, dtype=dtype, out=out
     )
+
+
+# nanmax
+@with_unsupported_dtypes({"2.0.1 and below": ("float16",)}, "torch")
+@infer_device
+@infer_dtype
+@handle_array_function
+@ivy.handle_out_argument
+@ivy.to_native_arrays_and_back
+@ivy.handle_device_shifting
+def nanmax(
+    a: ivy.Array,
+    /,
+    *,
+    axis: Optional[Union[Tuple[int], int]] = None,
+    keepdims: bool = False,
+) -> ivy.Array:
+    """
+    Compute the maximum value of an array while ignoring NaN values.
+
+    Parameters
+    ----------
+    a
+        Input array.
+    axis
+        Axis or axes along which to operate. Default is None.
+    keepdims
+        If True, retains reduced dimensions with size 1. Default is False.
+
+    Returns
+    -------
+    ret
+        The maximum value of the elements in `a`, ignoring NaN values.
+
+    Example:
+    >>> import ivy
+    >>> a = ivy.array([1.0, 2.0, ivy.nan, 4.0, 5.0])
+    >>> nanmax(a)
+    array(5., dtype=float32)
+    """
+    # Create a mask for NaN values in the input array
+    mask = ivy.isnan(a)
+
+    # Replace NaN values with negative infinity
+    masked_a = ivy.where(mask, -ivy.inf, a)
+
+    # Compute the maximum value of the masked array
+    backend = ivy.current_backend(a)
```

Comment:
backend doesn't need to be instantiated this way, check out other fns for convention

Reply:
@rishabgit, in local testing what could be the origin of this error? 
[Test Results - pytest_for_test_statistical_test_nanmax.pdf](https://github.com/unifyai/ivy/files/12833139/Test.Results.-.pytest_for_test_statistical_test_nanmax.pdf)